data/scripts.js

```diff
@@ -206,105 +213,104 @@ exports.BattleScripts = {
 			}
 		}
 
-		if (target === false) {
-			this.attrLastMove('[notarget]');
-			this.add('-notarget');
-			if (move.target === 'normal') pokemon.isStaleCon = 0;
-			return false;
-		}
+		let noTarget = target === false;
 
-		let targets = pokemon.getMoveTargets(move, target);
+		if (!noTarget) {
```

Comment:
Why isn't this also an early return?

Reply:
You're still indenting so much code by a ton.

Reply:
The whole reason I did it this way is because having to store the return value of a function in the same place every time that function is called no matter where it’s called from makes early returns annoying when the function returns in 9 places and dangerous when someone adds another early return later and doesn’t store the value. The alternative is to rename this function `useMoveInner` or something like that and make a new `useMove` that handles `useMoveInner`’s return values. That lets you have as many early returns as you want in the inner function, but pits a little extra strain on the call stack. I asked which of these approaches I should do twice in dev and got no response...

Reply:
I think it'd be worth it to have more functions.

Reply:
Yeah, in general readability takes priority over being nice to the call stack.